File: java/azure-storage-blob-perfstress/src/main/java/com/azure/storage/blob/perfstress/UploadBlockBlobTest.java
package com.azure.storage.blob.perfstress;

import java.io.IOException;

import com.azure.perfstress.RandomFlux;
import com.azure.perfstress.RandomStream;
import com.azure.perfstress.SizeOptions;
import com.azure.storage.blob.perfstress.core.RandomBlobTest;

import reactor.core.publisher.Mono;

public class UploadBlockBlobTest extends RandomBlobTest<SizeOptions> {
    public UploadBlockBlobTest(SizeOptions options) {
        super(options);
    }

    @Override
    public void Run() {
        try {
            _blockBlobClient.upload(RandomStream.create(Options.Size), Options.Size);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public Mono<Void> RunAsync() {
        return _blockBlobAsyncClient.upload(RandomFlux.create(Options.Size), Options.Size).then();
    }
}

File: net/Azure.Storage.Blobs.PerfStress/UploadBlockBlobTest.cs
using Azure.Storage.Blobs.PerfStress.Core;
using Azure.Storage.Blobs.Specialized;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Storage.Blobs.PerfStress
{
    public class UploadBlockBlobTest : RandomBlobTest<SizeOptions>
    {
        public UploadBlockBlobTest(SizeOptions options) : base(options)
        {
        }

        public override void Run(CancellationToken cancellationToken)
        {
            using var stream = RandomStream.Create(Options.Size);
            BlockBlobClient.Upload(stream, cancellationToken: cancellationToken);
        }

        public override async Task RunAsync(CancellationToken cancellationToken)
        {
            using var stream = RandomStream.Create(Options.Size);
            await BlockBlobClient.UploadAsync(stream, cancellationToken: cancellationToken);
        }
    }
}
